tests/test_download.py

```diff
@@ -96,6 +104,87 @@ def test_download_service(
         assert response.status_code == 400, f"{response.data}"
 
 
+@pytest.mark.integration
+@pytest.mark.parametrize(
+    "userdata,erc20_enterprise",
```

Comment:
I'm almost ready to approve, in the meantime I noticed all the parametrisations are copied from the single-order test. Do we need them in both places? Are they properly used in both tests? Maybe we can keep them in just one place, since there's no point in testing the same functionality multiple times, in 8 different tests.

Reply:
agreed